web-server/src/components/TopLevelLogicComponent.tsx

```diff
@@ -1,8 +1,12 @@
 import { FC } from 'react';
 
 import { useImageUpdateStatusWorker } from '@/hooks/useImageUpdateStatusWorker';
+import { getGithubRepoStars } from '@/slices/app';
+import { useDispatch } from '@/store';
 
 export const TopLevelLogicComponent: FC = () => {
+  const dispatch = useDispatch();
+  dispatch(getGithubRepoStars());
```

Comment:
BIG NO! This will call this function every time it re-renders.

Use a `useEffect`. It will only call the thunk on component-mounting (first-time render only)

Reply:
@e-for-eshaan Thnks